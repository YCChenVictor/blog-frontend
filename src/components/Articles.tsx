import * as React from "react";

type Articles = {
  url: string;
  content: string;
}[];

const Articles = ({ articles }: { articles: Articles }) => {
  // Group articles by their category
  const groupedArticles = Object.values(articles).reduce(
    (acc, value) => {
      const category = value.url.split("/")[1];
      if (category == "in-progress") {
        return acc;
      }
      if (!acc[category]) {
        acc[category] = [];
      }
      acc[category].push(value);
      return acc;
    },
    {} as Record<string, Articles>,
  );

  const list = Object.keys(groupedArticles).map((category, index) => (
    <div key={index} className="mb-6">
      <h2 className="font-bold text-gray-800 mb-4">{category}</h2>{" "}
      {/* Display category */}
      <div className="space-y-4">
        {groupedArticles[category].map((article, idx) => {
          const title = article.url.split("/")[2];
          return (
            <div className="flex">
              <div
                key={idx}
                className="p-4 bg-gray-600 rounded-lg shadow-md hover:bg-gray-200"
              >
                <a href={article.url} className="text-black hover:underline">
                  {title}
                </a>
              </div>
              <div>parent</div>
              <div>children</div>
            </div>
          );
        })}
      </div>
    </div>
  ));

  return (
    <div>
      <h1>Articles</h1>
      {list}
    </div>
  );
};

export default Articles;
