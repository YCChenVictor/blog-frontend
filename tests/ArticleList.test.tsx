import * as React from "react";
import * as renderer from "react-test-renderer";
import Articles from "../src/components/Articles";

describe("ArticleList", () => {
  it("renders ArticleList component without crashing", () => {
    const articles = [
      { url: "http://example.com/1", content: "Category 1" },
      { url: "http://example.com/2", content: "Category 2" },
    ];

    const component = renderer.create(<Articles articles={articles} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
