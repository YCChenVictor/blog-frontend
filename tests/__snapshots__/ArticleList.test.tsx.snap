// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders ArticleList component without crashing 1`] = `
<div>
  <h1>
    Articles
  </h1>
  <div
    className="mb-6"
  >
    <h2
      className="font-bold text-gray-800 mb-4"
    />
     
    <div
      className="space-y-4"
    >
      <div
        className="p-4 bg-gray-600 rounded-lg shadow-md hover:bg-gray-200"
      >
        <a
          className="text-black hover:underline"
          href="http://example.com/1"
        >
          example.com
        </a>
      </div>
      <div
        className="p-4 bg-gray-600 rounded-lg shadow-md hover:bg-gray-200"
      >
        <a
          className="text-black hover:underline"
          href="http://example.com/2"
        >
          example.com
        </a>
      </div>
    </div>
  </div>
</div>
`;
